trunk/cli/tests/integration_tests.rs

```diff
@@ -33,7 +36,61 @@ fn build_extension() -> Result<(), Box<dyn std::error::Error>> {
     cmd.arg(output_dir.clone());
     cmd.assert().code(0);
     assert!(
-        std::path::Path::new(format!("{output_dir}/test_extension-0.0.0.tar.gz").as_str()).exists()
+        std::path::Path::new(format!("{output_dir}/test_pgx_extension-0.0.0.tar.gz").as_str())
+            .exists()
+    );
+    // delete the temporary file
+    std::fs::remove_dir_all(output_dir)?;
+
+    Ok(())
+}
+
+#[test]
+fn build_c_extension() -> Result<(), Box<dyn std::error::Error>> {
+    let mut rng = rand::thread_rng();
+    let output_dir = format!("/tmp/pg_tle_test_{}", rng.gen_range(0..1000000));
+
+    let current_file_path = Path::new(file!()).canonicalize().unwrap();
+    // Example of a C extension
+    let repo_url = "https://github.com/aws/pg_tle.git";
+    // clone and checkout ref v1.0.3
+    let repo_dir_path = current_file_path.parent().unwrap().join("pg_tle");
+    let repo_dir = PathBuf::from(repo_dir_path);
+    if repo_dir.exists() {
+        fs::remove_dir_all(&repo_dir.clone()).unwrap();
+    }
+    let repo = Repository::clone(repo_url, &repo_dir).unwrap();
+    let refname = "v1.0.3";
+    let (object, reference) = repo.revparse_ext(refname).expect("Object not found");
+    repo.checkout_tree(&object, None)
+        .expect("Failed to checkout");
+    match reference {
+        // gref is an actual reference like branches or tags
+        Some(gref) => repo.set_head(gref.name().unwrap()),
+        // this is a commit, not a reference
+        None => repo.set_head_detached(object.id()),
+    }
+        .expect("Failed to set HEAD");
+
+    // Construct a path relative to the current file's directory
+    let mut extension_path = std::path::PathBuf::from(file!());
+    extension_path.pop(); // Remove the file name from the path
+    extension_path.push("pg_tle");
+
+    let mut cmd = Command::cargo_bin(CARGO_BIN)?;
+    cmd.arg("build");
+    cmd.arg("--path");
+    cmd.arg(extension_path.as_os_str());
+    cmd.arg("--output-path");
+    cmd.arg(output_dir.clone());
+    cmd.arg("--version");
+    cmd.arg("1.0.3");
+    cmd.arg("--name");
+    cmd.arg("pg_tle");
+    cmd.assert().code(0);
```

Comment:
this test clones pg_tle, then does `trunk build --path /to/pg_tle --version 1.0.3 --name pg_tle` and checks a trunk package was created.

TODO: test trunk install from this package and also from pgx-built package